From 358615f416a8f3085a63c03a55564f71946083d1 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 20 Jan 2016 05:35:11 +0000
Subject: [PATCH 016/117] tests: Include sys/select.h

Used in compliance with POSIX 2001/2008

Fixes errors e.g.
error: implicit declaration of function 'select'

and helps with missing definitions of FD_* defines

v2: conditionally include sys/select.h, include in every test where
needed.

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Reviewed-by: Thierry Reding <thierry.reding@gmail.com> (v1)
Reviewed-by: Emil Velikov <emil.l.velikov@gmail.com>
---
 configure.ac                     | 2 +-
 tests/kms/kms-steal-crtc.c       | 3 +++
 tests/kms/kms-universal-planes.c | 3 +++
 tests/modetest/modetest.c        | 3 +++
 tests/vbltest/vbltest.c          | 3 +++
 5 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index a09be61..4635d18 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,7 +53,7 @@ AC_USE_SYSTEM_EXTENSIONS
 AC_SYS_LARGEFILE
 AC_FUNC_ALLOCA
 
-AC_CHECK_HEADERS([sys/mkdev.h sys/sysctl.h])
+AC_CHECK_HEADERS([sys/mkdev.h sys/sysctl.h sys/select.h])
 
 # Initialize libtool
 LT_PREREQ([2.2])
diff --git a/tests/kms/kms-steal-crtc.c b/tests/kms/kms-steal-crtc.c
index 2f7f327..497772e 100644
--- a/tests/kms/kms-steal-crtc.c
+++ b/tests/kms/kms-steal-crtc.c
@@ -31,6 +31,9 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#ifdef HAVE_SYS_SELECT_H
+#include <sys/select.h>
+#endif
 
 #include <drm_fourcc.h>
 
diff --git a/tests/kms/kms-universal-planes.c b/tests/kms/kms-universal-planes.c
index 9151231..d8e5fc4 100644
--- a/tests/kms/kms-universal-planes.c
+++ b/tests/kms/kms-universal-planes.c
@@ -32,6 +32,9 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#ifdef HAVE_SYS_SELECT_H
+#include <sys/select.h>
+#endif
 
 #include <drm_fourcc.h>
 #include "xf86drm.h"
diff --git a/tests/modetest/modetest.c b/tests/modetest/modetest.c
index f665240..b8aa94b 100644
--- a/tests/modetest/modetest.c
+++ b/tests/modetest/modetest.c
@@ -55,6 +55,9 @@
 #include <errno.h>
 #include <sys/poll.h>
 #include <sys/time.h>
+#ifdef HAVE_SYS_SELECT_H
+#include <sys/select.h>
+#endif
 
 #include "xf86drm.h"
 #include "xf86drmMode.h"
diff --git a/tests/vbltest/vbltest.c b/tests/vbltest/vbltest.c
index 4475b49..97dd44d 100644
--- a/tests/vbltest/vbltest.c
+++ b/tests/vbltest/vbltest.c
@@ -37,6 +37,9 @@
 #include <errno.h>
 #include <sys/poll.h>
 #include <sys/time.h>
+#ifdef HAVE_SYS_SELECT_H
+#include <sys/select.h>
+#endif
 
 #include "xf86drm.h"
 #include "xf86drmMode.h"
-- 
2.7.4

